smart-contracts/assembly/contracts/FT/__tests__/WMAS.spec.ts

```diff
@@ -0,0 +1,126 @@
+import {
+  Address,
+  changeCallStack,
+  resetStorage,
+  setDeployContext,
+  mockBalance,
+  mockTransferredCoins,
+} from '@massalabs/massa-as-sdk';
+import { Args, u256ToBytes } from '@massalabs/as-types';
+import { balanceOf, constructor, deposit, withdraw } from '../WMAS';
+import { u256 } from 'as-bignum/assembly';
+
+// address of the contract set in vm-mock. must match with contractAddr of @massalabs/massa-as-sdk/vm-mock/vm.js
+const contractAddr = 'AS12BqZEQ6sByhRLyEuf0YbQmcF2PsDdkNNG1akBJu9XcjZA1eT';
+const user1Address = 'AU12UBnqTHDQALpocVBnkPNy7y5CndUJQTLutaVDDFgMJcq5kQiKq';
+const user2Address = 'AU12BqZEQ6sByhRLyEuf0YbQmcF2PsDdkNNG1akBJu9XcjZA1e8';
+const user3Address = 'AUDeadBeefDeadBeefDeadBeefDeadBeefDeadBeefDeadBOObs';
+
+const TOKEN_NAME = 'TOKEN_NAME';
+const TOKEN_SYMBOL = 'TKN';
+const DECIMALS: u8 = 9;
+const TOTAL_SUPPLY = u256.Zero;
+
+const amount = 1_000_000_000_000;
+const storageCost = computeStorageCost(new Address(user2Address));
+const amountMinusStorageCost = amount - storageCost;
+const tooLargeAmount = 2 * amount;
+
+function switchUser(user: string): void {
+  changeCallStack(user + ' , ' + contractAddr);
+}
+
+function computeStorageCost(receiver: Address): u64 {
+  let cost = 0;
+  cost = 400_000;
+  cost += 100_000 * (7 + receiver.toString().length);
+  cost += 3_200_000;
+  return cost;
+}
```

Comment:
Why not using the computeStorageCost of the smart contract ?

Reply:
Because it's an internal function with regular arguments. But I could export it and pass arguments as binaryArgs if you think it is better

Reply:
you can export it without changing the arguments. it will be usable in tests, but not by SC call

Reply:
Okay